Automattic/wp-calypso
# Happychat: Add visual chat status notifications

Depends on #11381.

By refactoring to a middleware, happychat can update UI when connection and message events happen.

- [x] Badge icon when a message is received while happychat is minimize
- [x] ~~Display messaging to user when server connection is down~~ **Moved to #11821**
- [x] ~~Display messaging to user when HE drops chat connection~~ **Moved to #11821**


### How to test

1. Log in to the staging HUD and make yourself available
1. Start a chat from Calypso
1. Minimize the Happychat popup
1. Send a message as the operator
1. Check that the badge icon appears in the bottom corner of the sidebar:

    ![screen shot 2017-02-14 at 11 52 53 am](https://cloud.githubusercontent.com/assets/416133/22911586/278c8300-f2ac-11e6-8f4c-9b2df4d58f64.png)

1. The badge icon should disappear when the button is clicked
